inbm/dispatcher-agent/dispatcher/dispatcher_class.py

```diff
@@ -731,14 +732,17 @@ def handle_updates(dispatcher: Any,
             dispatcher._send_result(f"Error parsing schedule manifest: {str(e)}", request_id)
             return
         
-        # TODO: Change single and repeated to add the schedules to the scheduler DB
+        # Add schedules to the database
         if schedule.single_scheduled_requests or schedule.repeated_scheduled_requests:
-            sql_lock.acquire()
-            for single_task in schedule.single_scheduled_requests:
-                SqliteManager().create_task(single_task)
-            for repeated_task in schedule.repeated_scheduled_requests:
-                SqliteManager().create_task(repeated_task)
-            sql_lock.release()
+            with sql_lock:            
+                def process_scheduled_requests(scheduled_requests: Sequence[Schedule]):
+                    sqliteManager = SqliteManager()
+                    for requests in scheduled_requests:
+                        sqliteManager.create_schedule(requests)
```

Comment:
**<h2>$\color{#FFA500}Improvements: Code Robustness$</h2>**
**Suggestion:** Consider using context manager for SQLite connection to ensure that resources are properly managed and connections are closed after operations are completed. This enhances the reliability and maintainability of the database operations.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
                    with SqliteManager() as sqliteManager:
                        for requests in scheduled_requests:
                            sqliteManager.create_schedule(requests)
```

Reply:
Removed the destructor so this is no longer valid.